backend/app/controllers/spree/admin/orders_controller.rb

```diff
@@ -166,7 +165,11 @@ def order_params
       end
 
       def load_order
-        @order = Spree::Order.includes(:adjustments).find_by!(number: params[:id])
+        begin
+          @order = Spree::Order.includes(:adjustments).find_by!(number: params[:id])
+        rescue ActiveRecord::RecordNotFound
+          resource_not_found(flash_class: Spree::Order, redirect_url: admin_orders_path)
+        end
```

Comment:
Similarly to my other comment, we can avoid unnecessary git churn by rescuing this at the method level:

```ruby
def load_order
  @order = Spree::Order.includes(:adjustments).find_by!(number: params[:id])
  authorize! action, @order
rescue ActiveRecord::RecordNotFound
  resource_not_found(flash_class: Spree::Order, redirect_url: admin_orders_path)
end
```